standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -1850,8 +1819,6 @@ private long partsFoundForPartitions(
           }
           Deadline.checkTimeout();
         }
-      } catch (Exception e) {
-        throwMetaOrRuntimeException(e);
```

Comment:
Is this exception handling already happens in the callers, or we end up throwing exceptions which were wrapped before, and will not be wrapped after this change?

Reply:
The exception handling has aready happened in the caller,  it's allowing to throw MetaException or RuntimeException.
The catch-clause is to catch the exception of releasing query with try-with-resources,  as it throws the  `Exception`, which should be transformed to `MetaException` or `RuntimeException`. This is what the `QueryWrapper` wants to solve,  we do not need this after the change.